spec/design/skel/Thread_H.thy

```diff
@@ -43,6 +43,9 @@ requalify_consts
 
 end
 
+definition awaken_def:
+"awaken \<equiv> whileLoop (\<lambda>_ s. the (fun_of_m releaseQNonEmptyAndReady s)) (\<lambda>_. awaken_body) ()"
```

Comment:
Can we line up the type of `funOfM` in Haskell with the one in Isabelle such that they match? Then we wouldn't need the manual translation here, but could restrict it to `whileLoop` and `funOfM` only.

(If `funOfM` is used in other places in Haskell as well, a different function could work as well)